### PR TITLE
I189 separate duplicationcheck

### PR DESCRIPTION
**Pull request by mataszilaitis**
1. Added JuniorTest, JuniorApp, OriginalTest and OriginalApp codebase and duplication costs to the appsettings.json file
2. Added two (_TestDuplicationCheck,AppDuplicationCheck_) and removed one (_DuplicationCheck_) check in `Feature.cs` and `FeatureDetailsStorage.cs`.
3. Two duplication checks are now running instead of one, DuplicationCheck isn't a check class by itself anymore. It is controlled by `TestDuplicationCheck` and `AppDuplicationCheck`
4. Still not sure about the correct weightings and where to store some of the evidence variables.

### CODE REVIEW

File: YoCode/DuplicationCheck.cs [hunk 1]
```diff
@@ -29,25 +29,18 @@ internal class DuplicationCheck
...
 
             this.dir = dir;
             this.dupFinder = dupFinder;
+            this.fileNameChecked = fileNameChecked;
 
             modifiedSolutionPath = Path.Combine(dir.ModifiedTestDirPath, fileNameChecked);
 
             try
             {
-                ExecuteTheCheck();
```
Comment (mataszilaitis) [inline YoCode/DuplicationCheck.cs]: also not sure what I need to put in here, since I don't want the DuplicationCheck to run automatically anymore.

Comment (jogobom) [reply YoCode/DuplicationCheck.cs]: Move the whole try/catch to `PerformDuplicationCheck`. It isn't needed in the constructor any more but it probably is needed in the place that code moved to, for the same reasons we had it here originally.

**Comment by jogobom** [comment]
I think it would be worth trying to resolve the merge conflicts before reviewing this. Update this branch from master, then request a review. That will allow us to check that the merge result is correct before merging back to master.

**Comment by jogobom** [comment]
It's a bit difficult to review this now, github really doesn't seem to do a good job with PRs where master has been merged in after the PR was created. I just got the code in Visual Studio and looked at it, rather than trying to use github's UI for the review.

**Pull request merged**